htlcswitch/switch.go

```diff
@@ -353,34 +340,44 @@ func (s *Switch) ProcessContractResolution(msg contractcourt.ResolutionMsg) erro
 	return nil
 }
 
-// SendHTLC is used by other subsystems which aren't belong to htlc switch
-// package in order to send the htlc update.
-func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID,
-	htlc *lnwire.UpdateAddHTLC,
-	deobfuscator ErrorDecrypter) ([sha256.Size]byte, error) {
+// GetPaymentResult returns the the result of the payment attempt with the
+// given paymentID. This method blocks until the result is available, or an
+// error is encountered. If the paymentID is unknown, ErrPaymentIDNotFound will
+// be returned.
+func (s *Switch) GetPaymentResult(paymentID uint64) (PaymentResult, error) {
```

Comment:
seems we should move `removePendingPayment` from local dispatch to this method, o/w there's an ordering where the pending payment has already been removed, but `GetPaymentResult` hasn't been called yet